### PR TITLE
Add boost dependencies to build file

### PR DESCRIPTION
**Pull request by AGaliuzov**
Add compilation flags

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,30 +1,67 @@
...
-      all {
-          linker.args "-lpthread"
+  binaries {
+    all {
+      if (buildType == buildTypes.debug) {
+        cppCompiler.args "-g"
       }
+
+      cppCompiler.args "-Wextra"
+      cppCompiler.args "-Wall"
```
Comment (malirod) [inline build.gradle]: @AGaliuzov, `Wextra` = `Wall` + some additional checks. Thus `Wall` is redundant.

File: build.gradle [hunk 2]
```diff
@@ -1,30 +1,67 @@
...
-    binaries {
-      all {
-          linker.args "-lpthread"
+  binaries {
+    all {
+      if (buildType == buildTypes.debug) {
+        cppCompiler.args "-g"
       }
+
+      cppCompiler.args "-Wextra"
```
Comment (malirod) [inline build.gradle:16]: @AGaliuzov, please add `-std=c++11` to have c++11 explicitly stated

File: build.gradle [hunk 3]
```diff
@@ -1,30 +1,67 @@
...
+        cpp {
+          source {
+            srcDirs += "thirdparty/gtest"
+            srcDirs += "src"
+            srcDirs += "test"
+            include "**/*.cc"
+          }
+          exportedHeaders {
+            srcDirs += "thirdparty/gtest"
+            srcDirs += "src"
```
Comment (malirod) [inline build.gradle]: @AGaliuzov, why "src" is included here?

File: build.gradle (hunk 2 above)
Comment (AGaliuzov) [reply build.gradle:16]: @malirod how about `pedantic`

Comment (AGaliuzov) [reply build.gradle:16]: do we need it?

Comment (malirod) [reply build.gradle:16]: @AGaliuzov, I've never used this mode. Propose to not be so fanatic.

Comment (malirod) [reply build.gradle:16]: @AGaliuzov, need what? I propose to drop Wall and use Wextra

File: build.gradle (hunk 3 above)
Comment (AGaliuzov) [reply build.gradle]: @malirod have you tried to include `logger.h` without this src dir here?

File: build.gradle (hunk 2 above)
Comment (AGaliuzov) [reply build.gradle:16]: @malirod need `pedantic`

File: build.gradle (hunk 3 above)
Comment (malirod) [reply build.gradle]: @AGaliuzov, nope. Seems you're right. In general I still don't like presence of exportedHeaders for the single executable.

File: build.gradle (hunk 2 above)
Comment (malirod) [reply build.gradle:16]: @AGaliuzov, if you insist, then we can add pedantic flag.

File: build.gradle (hunk 3 above)
Comment (AGaliuzov) [reply build.gradle]: @malirod actually it is expected - how the gradle should know where header is?
There are some rules we have to follow to allow gradle to locate the headers : Include directory should be located near the sources

Comment (malirod) [reply build.gradle]: @AGaliuzov, seems you're right.
By the way. seems it possible to write all paths in one line. See here https://discuss.gradle.org/t/c-plugin-exportedheaders-what-does-it-do/5041

I just don't get the idea of the exportedHeaders section. In all snippets i've seend before this section was related to the library. Exported means, how I've understood, that these headers are exported by library, and then executable just uses the lib. But here we have a single executable and still use this section.

Comment (AGaliuzov) [reply build.gradle]: @malirod Indeed. It confuse.
I know about all paths in one. I'm working on additional commit for this PR

**Comment by AGaliuzov** [comment]
@malirod see https://github.com/malirod/boost-log/pull/3/commits/1e6fbf57d868a0376c7642dc651e69a222dadf02

**Pull request merged**